muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/workspace/rest/WorkspaceRESTService.java

```diff
@@ -1075,6 +1076,16 @@ public Response listWorkspaceStaffMembers(@PathParam("ID") Long workspaceEntityI
       
       if (user != null) {
         UserEntity userEntity = userEntityController.findUserEntityByUser(user);
+        
+        // #3111: Workspace staff members should be limited to teachers only. A better implementation would support specified workspace roles
+        
+        if (workspaceEntity != null) {
```

Comment:
Unnecessary check (checked above), also makes the control flow a bit more complex to understand (should the workspaceStaffMembers.add be inside this if clause?).